packages/client/hmi-client/src/components/workflow/scenario-templates/counterfactual/tera-counterfactual-template.vue

```diff
@@ -0,0 +1,154 @@
+import { useProjects } from '@/composables/project';
+import Dropdown from 'primevue/dropdown';
+import _ from 'lodash';
+import { getModelConfigurationById } from '@/services/model-configurations';
+import { getInterventionPoliciesForModel, getModel, getModelConfigurationsForModel } from '@/services/model';
+import horizon from '@/assets/svg/template-images/horizon-thumbnail.svg';
+import MultiSelect from 'primevue/multiselect';
+import { ScenarioHeader } from '../base-scenario';
+import { CounterfactualScenario } from './counterfactual-scenario';
+import TeraScenarioTemplate from '../tera-scenario-template.vue';
+
+const props = defineProps<{
+	scenario: CounterfactualScenario;
+}>();
+const emit = defineEmits(['save-workflow']);
+
+const header: ScenarioHeader = Object.freeze({
+	title: 'Necessary or sufficient analysis template',
+	question: 'How likely would we arrive in a different state, if specific choices had been made differently?',
+	description:
+		'Requires data to calibrate against, intervention points in the past and binary test about the end state. This is a type of counterfactual analysis.',
+	examples: [
+		'Would we have 10% fewer cases now if vaccinations had been 5% higher a month ago?',
+		'Would the storm winds have been at least 10mph less if the ocean temperature were 1 degree lower?'
+	]
+});
+
+const allModelOptions = computed<ProjectAsset[]>(() => useProjects().getActiveProjectAssets(AssetType.Model));
+const allDatasetOptions = computed<ProjectAsset[]>(() => useProjects().getActiveProjectAssets(AssetType.Dataset));
+const selectedModelId = computed<string>(() => props.scenario.getModelId());
+const selectedModelConfigurationId = computed<string>(() => props.scenario.getModelConfigId());
+const isModelSelected = computed<boolean>(() => !_.isEmpty(selectedModelId.value));
+const model = ref<Model | null>(null);
+const modelConfiguration = ref<ModelConfiguration>();
+const allModelConfigOptions = ref<ModelConfiguration[]>([]);
+const allInterventionPolicyOptions = ref<InterventionPolicy[]>([]);
+const isFetchingModelData = ref<boolean>(false);
+const modelStateAndObsOptions = ref<string[]>([]);
+
+watch(
+	() => selectedModelId,
+	async () => {
+		if (selectedModelId.value) {
+			isFetchingModelData.value = true;
+			model.value = await getModel(selectedModelId.value);
+			allModelConfigOptions.value = await getModelConfigurationsForModel(selectedModelId.value);
+			allInterventionPolicyOptions.value = await getInterventionPoliciesForModel(selectedModelId.value);
+
+			// Set model state and obs:
+			if (model.value) {
+				// States:
+				const modelStates = model.value.model.states;
+				modelStates.forEach((state) => modelStateAndObsOptions.value.push(state.id));
+				// Obs:
+				const modelObs = model.value.semantics?.ode.observables;
+				if (modelObs) {
+					modelObs.forEach((obs) => modelStateAndObsOptions.value.push(obs.id));
+				}
+			}
+		}
+
+		isFetchingModelData.value = false;
+	},
+	{ deep: true }
```

Comment:
You either
- do not need a deep watcher because the value being watched is just a string, or
- you might be watching the wrong variable

Reply:
Good catch!